client/kaia_client.go

```diff
@@ -433,11 +484,15 @@ func (ec *Client) PendingTransactionCount(ctx context.Context) (uint, error) {
 // blocks might not be available.
 func (ec *Client) CallContract(ctx context.Context, msg kaia.CallMsg, blockNumber *big.Int) ([]byte, error) {
 	var hex hexutil.Bytes
-	err := ec.c.CallContext(ctx, &hex, "kaia_call", toCallArg(msg), toBlockNumArg(blockNumber))
-	if err != nil {
-		return nil, err
+	kaiaCallErr := ec.c.CallContext(ctx, &hex, "kaia_call", toCallArg(msg), toBlockNumArg(blockNumber))
+	if kaiaCallErr == nil {
+		return hex, nil
 	}
-	return hex, nil
+	auctionCallErr := ec.c.CallContext(ctx, &hex, "auction_call", toCallArg(msg), toBlockNumArg(blockNumber))
+	if auctionCallErr == nil {
+		return hex, nil
+	}
+	return nil, errors.Join(kaiaCallErr, auctionCallErr)
```

Comment:
If the user makes mistake (e.g. transaction revert or wrong format), then you always fall back to auction_call.
1. It incurs 2 http requests.
2. The error message would be confusing as it contains auction_call not found error. e.g. `transaction reverted, auction_call rpc not supported`

Reply:
If the primary user of Auction APIs is the Auctioneer, the Auctioneer knows you must use auction_ namespace. So I don't think you need this fallback mechanism throughout kaia_client.go